File: 03-week-3--additional-practice/04-bonus-5--consecutive-substrings/ruby/consecutive_substrings.rb
def consecutive_substrings(string)
  subString = []
  string.split('').each_with_index do |char, index|
    subString << char
    frag = char
    string[index + 1..-1].split('').each_with_index do |letter|
      frag += letter
      subString << frag
    end
  end
  subString
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: ['a', 'ab', 'abc', 'b', 'bc', 'c']"
  print "=> " 
  print consecutive_substrings('abc')

  puts

  puts "Expecting: ['a']"
  print "=> " 
  print consecutive_substrings('a')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: 03-week-3--additional-practice/04-bonus-5--consecutive-substrings/javascript/consecutive_substrings.js
function consecutiveSubstrings(string) {
  const arrayString = string.split('')
  const subString = []
  arrayString.forEach((char, index) => {
    subString.push(char)
    let frag = char
    arrayString.slice(index + 1).forEach((letter) => {
      frag = frag + letter
      subString.push(frag)
    })
  })
  return subString
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: ['a', 'ab', 'abc', 'b', 'bc', 'c']");
  console.log("=>", consecutiveSubstrings('abc'));

  console.log("");

  console.log("Expecting: ['a']");
  console.log("=>", consecutiveSubstrings('a'));
}

module.exports = consecutiveSubstrings;

// Please add your pseudocode to this file
// And a written explanation of your solution
